File: app/src/main/java/funnybrain/hsu/idv/taipeipark/MainActivity.java
package funnybrain.hsu.idv.taipeipark;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import funnybrain.hsu.idv.taipeipark.restful.model.Park;

public class MainActivity extends AppCompatActivity implements ParkContract.View {
    private static final String TAG = "MainActivity";

    @BindView(R.id.list) RecyclerView recyclerView;

    private String scope = "resourceAquire";
    private String rid = "bf073841-c734-49bf-a97f-3757a6013812";
    private ParkAdapter parkAdapter;
    private List<Park> dataList = new ArrayList<>();

    private ParkContract.Presenter mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        mPresenter = new ParkPresenter(ParkRepository.Companion.getInstance(scope, rid), this);

        parkAdapter = new ParkAdapter(this, dataList);
        recyclerView.setHasFixedSize(true);
        recyclerView.setAdapter(parkAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
    }

    @Override
    protected void onResume() {
        super.onResume();
        mPresenter.loadTask(true);
    }

    @Override
    public void setPresenter(ParkContract.Presenter presenter) {
        mPresenter = presenter;
    }

    @Override
    public void showProgress(boolean show) {

    }

    @Override
    public void showPark(List<Park> parks) {
        Log.d(TAG, ">>>>> " + parks.size());
        dataList.clear();
        dataList.addAll(parks);
        parkAdapter.notifyDataSetChanged();
    }
}

File: app/src/main/java/funnybrain/hsu/idv/taipeipark/main/MainActivity.kt
package funnybrain.hsu.idv.taipeipark.main

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import funnybrain.hsu.idv.taipeipark.main.adapter.ParkAdapter
import funnybrain.hsu.idv.taipeipark.data.source.ParkRepository
import funnybrain.hsu.idv.taipeipark.data.model.Park
import org.jetbrains.anko.setContentView
import java.util.*

class MainActivity : AppCompatActivity(), MainContract.View {

    private val scope = "resourceAquire"
    private val rid = "bf073841-c734-49bf-a97f-3757a6013812"
    private var parkAdapter: ParkAdapter? = null
    private val dataList = ArrayList<Park>()

    private var mPresenter: MainContract.Presenter? = null

    lateinit var mainUi: MainActivityUi

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mainUi = MainActivityUi()
        mainUi.setContentView(this)

        parkAdapter = ParkAdapter(dataList)
        mainUi.list.adapter = parkAdapter

        mPresenter = MainPresenter(ParkRepository.getInstance(scope, rid), this)
    }

    override fun onResume() {
        super.onResume()
        mPresenter!!.loadTask(true)
    }

    override fun setPresenter(presenter: MainContract.Presenter) {
        mPresenter = presenter
    }

    override fun showProgress(show: Boolean) {
    }

    override fun showPark(parks: List<Park>) {
        dataList.clear()
        dataList.addAll(parks)
        parkAdapter!!.notifyDataSetChanged()
    }

    companion object {
        private val TAG = "MainActivity"
    }
}
